server/game/boxd_game.py

```diff
@@ -0,0 +1,351 @@
+class BoxdGame(object):
+
+    def __init__(self):
+        self.__players = []
+        self.__board = Board(10, 10)
+
+    def add_player(self, player_id):
+
+        if player_id in self.__players:
+            raise ValueError("Player already in game:  {}".format(player_id))
+
+        self.__players.append(player_id)
+
+    def remove_player(self, player_id):
+
+        if player_id not in self.__players:
+            raise ValueError("Player is not in game:  {}".format(player_id))
+
+        # find all of the boxes owned by player_id
+        to_void = []
+        for (point, owner_id) in self.__board.get_boxes():
+            if owner_id == player_id:
+                to_void.append(point)
+
+        # remove player_id from ownership
+        if __name__ == '__main__':
+            for box_corner in to_void:
+                self.__board.remove_box_owner(box_corner)
+
+        self.__players.remove(player_id)
+
+        # for now, leave all lines owned by player_id.  Those can stay
+
+    def get_players(self):
+        return self.__players
+
+    def claim_line(self, pt1, pt2, player_id):
+
+        try:
+            return self.__board.claim_edge(pt1, pt2, player_id)
+        except EdgeOwnedError:
+            # this presumably means another client snagged the line first
+            # do we need to tell the client to redraw a line, or just rely on the other client's successful claim msg?
```

Comment:
I really hope we find out that this is highly unlikely or never happens, but I think you're right that it will just be some UX decision (e.g., "Line was taken!  Re-draw.")